src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1741,11 +1741,13 @@ trait Contexts { self: Analyzer =>
       var selectors = tree.selectors
       @inline def current = selectors.head
       while ((selectors ne Nil) && result == NoSymbol) {
-        if (current.introduces(name.toTermName))
+        def sameName(name: Name, other: Name) = {
+          (name eq other) || (name ne null) && name.start == other.start
+        }
+        if (name == nme.WILDCARD && current.isLiteralWildcard || sameName(current.rename, name))
           result = qual.tpe.nonLocalMember( // #2733: consider only non-local members for imports
-            if (name.isTypeName) current.name.toTypeName else current.name
-          )
-        else if (!current.isWildcard && current.name == name.toTermName)
+            if (name.isTypeName) current.name.toTypeName else current.name)
+        else if (!current.isWildcard && sameName(current.name, name))
```

Comment:
We should to change this to keep using the newly introduced method, `ImportSelector#introduces`, along these lines.

```patch
-if (current.introduces(name.toTermName))
if (current.introduces(name))

- else if (!current.isWildcard && current.name == name.toTermName)
+ else if (!current.isWildcard && current.hasName(name)
```

```diff
+  private def sameName(name: Name, other: Name) =  (name eq other) || (name ne null) && name.start == other.start
+  def hasName(other: Name) = sameName(target, other)
    def introduces(target: Name) =
      if (target == nme.WILDCARD) isLiteralWildcard
-      else target != null && target == rename
+      else target != null && sameName(target, rename)
  }
```

Reply:
Thanks, could you take another look?